src/js/common/validator.js

```diff
@@ -1,122 +1,470 @@
+        "pattern": "schema-validator-pattern",
+        "propertyNames": "schema-validator-propertyNames",
+        "required": "schema-validator-required",
+        "then": "schema-validator-then",
+        "type": "schema-validator-type",
+        "uniqueItems": "schema-validator-uniqueItems"
+    };
 
-var Ajv = Ajv || require("ajv");
+    gpii.schema.validator.defaultAjvOptions = {
+        // Generate a complete list of errors rather than stopping on the first failure.
+        allErrors: true,
 
-fluid.registerNamespace("gpii.schema.validator.ajv");
+        // Ignore AJV's error messages.
+        messages: false,
 
-gpii.schema.validator.ajv.init = function (that) {
-    // We persist a single AJV instance so that we can take advantage of its automatic compiling and caching.
-    that.ajv = Ajv(that.options.validatorOptions); /* eslint new-cap: "off" */
+        // Increase the precision of "multipleOf" checks beyond integers.
+        // See https://github.com/epoberezkin/ajv#advanced-options
+        multipleOfPrecision: 10,
 
-    gpii.schema.validator.ajv.refreshSchemas(that);
-};
+        // Prevent invalid data (such as passwords) from being displayed in error messages
+        verbose: false
+    };
 
-/**
- *
- * @param that {Object} the validator component itself.
- * @param key {String} The key of the JSON Schema we are validating against.
- * @param content {Object} The JSON data to be validated.
- * @return `{Object}` sanitized validation errors, if there are any, or `undefined` if there are no validation errors.
- *
- */
-gpii.schema.validator.ajv.validate = function (that, key, content) {
-    // TODO:  Review with Antranig, this is meant to ensure that validation only takes place when already we have a valid schema.
-    if (that.model.schemas[key]) {
-        var contentValid = that.ajv.validate(key, content);
+    /**
+     *
+     * Validate material against a "GPII Schema System" schema.
+     *
+     * @param {Any} toValidate - The material to be validated.
+     * @param {Object} gssSchema - A GSS schema definition.
+     * @param {Object} ajvOptions - Optional arguments to pass to the underlying AJV validator.
+     * @return {Object} - An object that describes the results of validation.  The `isValid` property will be `true` if the data is valid, or `false` otherwise.  The `isError` property will be set to `true` if there are validation errors.
+     *
+     */
+    gpii.schema.validator.validate = function (toValidate, gssSchema, ajvOptions) {
+        ajvOptions = ajvOptions || gpii.schema.validator.defaultAjvOptions;
 
-        if (!contentValid) {
-            return (gpii.schema.validator.ajv.sanitizeValidationErrors(that, key, that.ajv.errors));
+        // Validate the GSS schema against the metaschema before proceeding.
+        var schemaValidationResults = gpii.schema.validator.validateSchema(gssSchema, ajvOptions);
+        if (fluid.get(schemaValidationResults, "isError")) {
```

Comment:
I imagine a simple schemaValidationResults.isError is good enough since the validator always guarantees to return an object

Reply:
Done.